src/core/annotation.js

```diff
@@ -2286,6 +2336,67 @@ class CircleAnnotation extends MarkupAnnotation {
     super(parameters);
 
     this.data.annotationType = AnnotationType.CIRCLE;
+
+    if (!this.appearance) {
+      // The default stroke color is black.
+      const strokeColor = this.color
+        ? Array.from(this.color).map(c => c / 255)
+        : [0, 0, 0];
+
+      // The default fill color is transparent.
+      let fillColor = null;
+      let interiorColor = parameters.dict.getArray("IC");
+      if (interiorColor) {
+        interiorColor = getRgbColor(interiorColor);
+        fillColor = interiorColor
+          ? Array.from(interiorColor).map(c => c / 255)
+          : null;
+      }
+
+      this._setDefaultAppearance({
+        xref: parameters.xref,
+        extra: `${this.borderStyle.width} w`,
+        strokeColor,
+        fillColor,
+        pointsCallback: (buffer, points) => {
+          // Circles are approximated by Bézier curves with four segments since
+          // there is no circle primitive in the PDF specification. For the
+          // control points distance, see https://stackoverflow.com/a/27863181.
+          const controlPointsDistance = (4 / 3) * Math.tan(Math.PI / (2 * 4));
```

Comment:
Nit: Maybe move this out of the callback-function, since it's a proper constant?

Reply:
Done in the new commit.